File: src/gestorAplicacion/servicios/CitaVacuna.java
/* Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
y Santiago Arboleda Acevedo */
package gestorAplicacion.servicios;

import gestorAplicacion.administracion.Vacuna;
import gestorAplicacion.personas.Paciente;
import gestorAplicacion.personas.Persona;

//Clase destinada a las citas que son de vacunas
public class CitaVacuna extends Cita{

    //Atributos
    private Vacuna vacuna;

    //Constructor
    public CitaVacuna(String fecha, Paciente paciente, Vacuna vacuna){
        super(null, fecha, paciente);
        this.vacuna=vacuna;
    }

    //Métodos

    //Método para buscar id del servicio y cambiar estado de pago
    @Override
    public void validarPago(Paciente paciente, long idServicio) {
        for (CitaVacuna citaVacuna :
                paciente.getHistoriaClinica().getHistorialVacunas()) {
            if (citaVacuna.getIdServicio() == idServicio)
                citaVacuna.setEstadoPago(true);
        }
    }

    //Getters y Setters
    public String mensaje(){
        return "del servicio de vacunas!";
    }
    public Vacuna getVacuna() {
        return vacuna;
    }

    public void setVacuna(Vacuna vacuna) {
        this.vacuna = vacuna;
    }



}

File: Python/src/gestor_aplicacion/servicios/cita_vacuna.py
# Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
# y Santiago Arboleda Acevedo
from src.gestor_aplicacion.servicios.cita import Cita


# Clase destinada a las citas que son de vacunas
class CitaVacuna(Cita):

    # Atributos y constructor
    def __init__(self, fecha, paciente, vacuna):
        super().__init__(paciente, None, fecha)
        self._vacuna = vacuna

    # Metodos

    # Metodo que imprime una descripcion del servicio
    def descripcion_servicio(self):
        return f"{self._id_servicio} - Vacuna ({self.vacuna.nombre})"

    # Métodos que busca el estado de pago de una cita médica y lo cambia
    # Recibe un objeto de la clase Paciente y un entero
    def confirmar_pago(self, paciente, id_servicio):
        for cita_vacuna in paciente.historia_clinica.historial_vacunas:
            if cita_vacuna.id_servicio == id_servicio:
                cita_vacuna.estado_pago = True
                break

    @property
    def vacuna(self):
        return self._vacuna

    @vacuna.setter
    def vacuna(self, vacuna):
        self._vacuna = vacuna

